framework/src/modules/http_api/controllers/delegates.js

```diff
@@ -183,6 +144,94 @@ DelegatesController.getForgingStatistics = async function(context, next) {
 	});
 };
 
+/**
+ * Gets a list of delegates based on query parameters
+ * @param {Object} filters - Query object
+ * @param {int} filters.limit - Limit applied to results
+ * @param {int} filters.offset - Offset value for results
+ * @param {object} options - Filter options
+ * @returns {Promise<*>}
+ * @private
+ */
+async function _getDelegates(filters, options) {
+	const delegates = await storage.entities.Account.get(
+		{ isDelegate: true, ...filters },
+		options
+	);
+
+	const [lastBlock] = await storage.entities.Block.get(
+		{},
+		{ sort: 'height:desc', limit: 1 }
+	);
+
+	const supply = lastBlock.height
+		? await channel.invoke('chain:calculateSupply', [lastBlock.height])
+		: 0;
+
+	return delegates.map(delegate => delegateFormatter(supply, delegate));
+}
+
+/**
+ * Gets a list forgers based on query parameters.
+ *
+ * @param {Object} filters - Query object
+ * @param {int} filters.limit - Limit applied to results
+ * @param {int} filters.offset - Offset value for results
+ * @returns {Promise<Array<object>>}
+ * @private
+ */
+async function _getForgers(filters) {
+	const [lastBlock] = await storage.entities.Block.get(
+		{},
+		{ sort: 'height:desc', limit: 1 }
+	);
+
+	const lastBlockSlot = slots.getSlotNumber(lastBlock.timestamp);
+	const currentSlot = slots.getSlotNumber();
+	const forgerKeys = [];
+	const round = slots.calcRound(lastBlock.height + 1);
+
+	const activeDelegates = await channel.invoke('chain:generateDelegateList', [
+		round,
+		null,
+		null,
+	]);
+
+	for (
+		let i = filters.offset + 1;
+		i <= ACTIVE_DELEGATES && i <= filters.limit + filters.offset;
+		i++
+	) {
+		if (activeDelegates[(currentSlot + i) % ACTIVE_DELEGATES]) {
+			forgerKeys.push(activeDelegates[(currentSlot + i) % ACTIVE_DELEGATES]);
+		}
+	}
+
+	const forgers = (await storage.entities.Account.get(
+		{ isDelegate: true, publicKey_in: forgerKeys },
+		{ limit: null }
+	))
+		.map(({ username, address, publicKey }) => ({
+			username,
+			address,
+			publicKey,
+			nextSlot: forgerKeys.indexOf(publicKey) + currentSlot + 1,
+		}))
+		.sort((prev, next) => prev.nextSlot > next.nextSlot);
```

Comment:
it's better to check if this order is correct or intended. (im not sure which order it was originally

Reply:
@limiaspasdaniel For reference that was the old logic 

https://github.com/LiskHQ/lisk/blob/5c7e2944383bf369f1343fba62c71e84de6f9664/modules/delegates.js#L883

Reply:
Double checked. The new code sorts the same way as sortBy.